dialogue-core/src/main/java/com/palantir/dialogue/core/Dialogue.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.google.errorprone.annotations.MustBeClosed;
+import com.palantir.dialogue.ConjureRuntime;
+
+public final class Dialogue {
```

Comment:
My dream is if we can make all the things people care about accessible from one top-level starting point, so they can just remember "the method is on Dialogue somewhere".

Reply:
I think if that's the case it'd be better to return a ClientFactory that uses the provided ClientPool so that no user of Dialogue has to know anything about something called a ClientPool.

Reply:
it's not so clear to me that this code should live in dialogue land rather than in wc?

Reply:
@uschi2000 are you referring to just this one entrypoint or the whole PR?

Reply:
I think the idea of a "singleton client factory", i.e., something that holds on to resources for the lifetime of a process, is something I'd expect in a server library, not a client library.

Reply:
I think if we try to make all our libraries good enough to use on their own, then when we pull them into WC they'll still be good. I actually think a client library that doesn't give you a convenient way to release resources when you want to seems broken.

To be clear this API doesn't know anything about the lifetime of a process, and there are no static singletons here.  it would still be up to witchcraft to make sure there's one client pool and witchcrafts responsibility to close it at shutdown time.

Reply:
`Dialogue` is designed to be a singleton, no?
Yes, it's a bug today that we don't close channels, but then why isn't it sufficient for this library to add `Channel#close`? I prefer hierarchical ownership (with delegating `.close()`) over complex "inside-out" resource management; is that not an option here?